### PR TITLE
Fix MDADM on reboot

### PR DESCRIPTION
**Pull request by obierlaire**
As explained in https://github.com/elastic/sdh-cloud/issues/11718 `mdadm` is ignored when rebooting.
In order to reproduce the issue you need at least two secondary disks, with Centos 

I'll test this PR by creating new AMIs with this PR

Please note that this does not have any impact on ansible-playbook direct install mode since we are not currently supporting more than one secondary disk (configured with `device_name`)

### CODE REVIEW

**Comment by obierlaire** [comment]
While testing this PR, I also discovered that centos8 does not even have `mdadm`, and so, if you try to start an instance based on this AMI with more than one secondary disk, they are not mounted properly (and docker uses the main disk 😞 ) 
This PR is also fixing this problem

**Comment by obierlaire** [comment]
For testing this, I've build AMIs for centos 7 and 8, ubuntu xenial and bionic. For each I spin up an instance (different generations in order to test legacy and nvme disks), and attached 3 disks (main + 2 secondary). Once up I checked the disk structure:
```
$ lsblk 

NAME         MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
nvme0n1      259:2    0    50G  0 disk  
└─nvme0n1p1  259:3    0    50G  0 part  /
nvme1n1      259:0    0    80G  0 disk  
└─md0          9:0    0 179.9G  0 raid0 
  ├─lxc-swap 253:0    0   3.7G  0 lvm   [SWAP]
  └─lxc-data 253:1    0 176.2G  0 lvm   /mnt/data
nvme2n1      259:1    0   100G  0 disk  
└─md0          9:0    0 179.9G  0 raid0 
  ├─lxc-swap 253:0    0   3.7G  0 lvm   [SWAP]
  └─lxc-data 253:1    0 176.2G  0 lvm   /mnt/data
```

Then I restarted the instance and run `lsblk` again

File: templates/format-drives.j2 [hunk 1]
```diff
@@ -44,8 +44,6 @@ else
 
   [[ -e ${MDADM_CONF} ]] && cp ${MDADM_CONF} ${MDADM_CONF}.backup
   install -Dv /dev/null ${MDADM_CONF}
-  echo DEVICE ${drives} | tee ${MDADM_CONF}
```
Comment (rhass) [inline templates/format-drives.j2:47]: How/where do we setup MDADM_CONF if we are no longer writing the file here?

Comment (obierlaire) [reply templates/format-drives.j2:47]: 2 lines later 😉 
`mdadm --examine --scan | tee -a ${MDADM_CONF}`

**Review by vaubarth — APPROVED** [review]

File: templates/format-drives.j2 (hunk 1 above)
Comment (rhass) [reply templates/format-drives.j2:47]: Ah, right!

**Review by rhass — APPROVED** [review]

**Pull request merged**